src/containers/Bookings.js

```diff
@@ -4,33 +4,79 @@ import Results from "../components/Results.js";
 import * as FakeBookings from "../data/fakeBookings.json";
 
 export default class Bookings extends Component {
-  constructor(props){
+  constructor(props) {
     super(props);
-      this.state= {results:[]};
-
+    this.state = { results: [], direction: 'asc', style: {} };
 
+    this.addNumberOfDays(FakeBookings);
   }
+
+
+  addNumberOfDays = (FakeBookings) => {
+    FakeBookings.map((item) => {
+      var d1 = new Date(item.checkInDate)
+      var d2 = new Date(item.checkOutDate)
+      item.numberOfDays = Math.round((d2 - d1) / (1000 * 60 * 60 * 24));
+
+    })
+
+  };
   
+
+
+
   searchId = () => {
-    this.setState({results:FakeBookings}) 
+    this.setState({ results: FakeBookings })
+
     console.info("to do!");
   };
   searchName = () => {
-    this.setState({results:FakeBookings}) 
+    this.setState({ results: FakeBookings })
     console.info("not to do!");
   };
-  sortBy = (colum)=>{
-    FakeBookings.sort((a,b)=>{if(a[colum]>b[colum]){return 1}})
-    this.setState({results:FakeBookings})
+
+
+  sortBy = (key, type) => {
+    this.setState({
+      results: FakeBookings.sort((a, b) => {
+        if (type === 'string') {
+          return this.state.direction === 'asc'
+            ? (a[key].toLowerCase() > b[key].toLowerCase())
+            : (a[key].toLowerCase() < b[key].toLowerCase())
+        } else {
+          return this.state.direction === 'asc'
+            ? (a[key] > b[key])
+            : (a[key] < b[key])
+        }
+      }
+      ),
+      direction: this.state.direction === 'asc'
+        ? 'desc'
+        : 'asc'
+    })
   }
-  
+  toggleSelected = (x) => {
```

Comment:
Having the selected state up in bookings might be a bit problem later - because you would toggle selected state for all items in the list, not just the one you selected. This state might be better further down the tree